library/insights_registration.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2018, Terry Jones <terry.jones@example.org>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: insights_registration
+
+short_description: This module registers the insights client
+
+description:
+    - This module will check the current registration status, unregister if needed,
+    and then register the insights client (and update the display_name if needed)
+
+options:
+    insights_name:
+        description:
+            - For now, this is just 'insights-client', but it could change in the future
+            so having it as a variable is just preparing for that
+        required: true
+    display_name:
+        description:
+            - When registering with insights an optional display_name can be configured
+        required: false
+
+author:
```

Comment:
Does it make sense to add a `state` parameter, accepting values `present` and `absent` (similar to `redhat_subscription`) so we can test that registration breadcrumbs have been removed from the local system.  I understand there may not be an unregister API call available yet, so that part could be added later.

Reply:
Along these lines, while testing I'm finding that while the breadcrumbs for a registered system exist in `/etc/insights-client/` the `insights_client.ini` has more recent changes.  I think a nice capability to add would be to for a re-register if the config file `mtime` is greater than that of the `machine-id`.

Reply:
I'll look into this!